### ISSUE TITLE
Same sprite in multiple objects

### ISSUE THREAD

**Issue by elewental** [whiteblXK/ObjectBuilder#9]
From: https://github.com/ottools/ObjectBuilder/issues/38

Using same sprite to different objects may help not making the client so heavy sometimes, perhaps is not something so impactful, and im not sure if everyone who do this way uses this reason, but I do sometimes.

And when I need add new sprites for only one of the objects, I must replace it by hand. Well I have a suggestion for this situation.

- Maybe if instead of the "you want to replace the current sprites", it appears "it was found more than one object with the same sprites, select this if you want to replace only this object id" and other option at the same window "ignore and replace all" (only in cases of really having multiple objects using same sprite id -- if not, keep the actual message/mode).

- As a result, if two objects have same sprite id, and you choose to replace only the 1 (the message "it was found more than one... Replace only this...) and not the 2, instead of replacing the 1 the same way it works now, it would work like if you're importing new sprites and replacing it automatically (it would do the work of remaking the object, once it is already configured), and then object 1 would have different sprite id from the 2.

Example:
![image](https://user-images.githubusercontent.com/52132367/156059327-5e1e6f67-4261-457f-9573-4282f7f56c9f.png)
![image](https://user-images.githubusercontent.com/52132367/156060422-e62da325-59cd-4df9-84ad-b0e9dd540a93.png)
![image](https://user-images.githubusercontent.com/52132367/156060910-4917f6e3-2c03-434a-8ded-aab5d5a13074.png)
![image](https://user-images.githubusercontent.com/52132367/156061090-6665a971-1896-4777-98b0-7ca53f3c97ea.png)

**Comment by punkice3407** [comment]
If I understand correctly, you are telling about Sprites Optimizer which is in Menu/Tools. It will remove all empty, unused and duplicated sprites at once

**Comment by elewental** [comment]
Not exactly.

Example:
![test4](https://user-images.githubusercontent.com/52132367/156479095-a0113931-b59d-4662-bbee-cc7dac85fb1c.gif)

For a random reason, I have 3 duplicated objects (100, 101 and 102). 
Each one is 2x1. Normally it would have 6 sprite ids for it, but I used one id equal in the 3 objects, so I have 4 sprite ids instead.
![image](https://user-images.githubusercontent.com/52132367/156480630-161e982c-d746-439a-b387-7dd6f18a8c87.png)

Then, if I want to **replace only the first one, the object id 100**, the sprite of 101 and 102 will be replaced too by issue:
![test5](https://user-images.githubusercontent.com/52132367/156479758-a780fc6e-2e48-46ad-8d25-5615ccd05ce7.gif)

The proposal is to create a option, for **instead of replacing the sprite ID that will change in all objects**, if you not choose "replace all", it will **import new sprites at the end of the spriteid list and then automatically replace it as another sprite id**, keeping the 101 and 102 configuration. 

![image](https://user-images.githubusercontent.com/52132367/156480257-456c7323-e4dc-41cb-bcaf-627267a04fec.png)

You see, If I choose to replace all, it will remain 4 sprite ids. (and keeping the issue shown at first gif)
![image](https://user-images.githubusercontent.com/52132367/156481695-0bef04d6-753b-44eb-ab64-d16ed64d9a4b.png)
![image](https://user-images.githubusercontent.com/52132367/156482063-c5a901ef-8a2d-4aa5-958b-b47eab47aae2.png)


But If I use the option I suggested, this will happen: (101 and 102 will still have a same spriteid, but will not share it with the 100).
![image](https://user-images.githubusercontent.com/52132367/156481432-de757e4a-fd7a-44fa-8ad4-111e0b0083cc.png)
![image](https://user-images.githubusercontent.com/52132367/156481545-1f4f1866-9c5f-4250-82b8-08910b347e4f.png)

**Comment by punkice3407** [comment]
Does the same problem occur with the 1x1 spritesheet?

**Comment by elewental** [comment]
It happens for all, when the same spriteid is used in more than 1 object, doesn't matter the width/weight or pattern, or the square the sprite is located.

**Comment by the-overdriven** [comment]
+1
This is a big problem especially when editing ground tiles which share same sprites.
Currently if you want to change a sprite only for one tile it will change it for all other tiles too!

When importing .png spritesheets it should always create new sprites, never replace old sprites!
When importing .obd it already does that, so why not do the same on .png import?

**Comment by elewental** [comment]
> When importing .png spritesheets it should always create new sprites, never replace old sprites!

> When importing .obd it already does that, so why not do the same on .png import?

I disagree, the function of replace action is to replace, if you want to import new sprites you could just use the import tool or drag the .png into a blank object. 

If you create an object with no sprites, set the correct size and then move a .png file it will add the sprites importing it.

![test13](https://user-images.githubusercontent.com/52132367/156945967-4353e15e-bd99-4ac5-a67d-f7a750ed31ea.gif)
